### ISSUE TITLE
Allow minimum severity for non-zero exit to be set via flag

### ISSUE THREAD

**Issue by bendrucker** [terraform-linters/tflint#1556]
## Introduction

Users may want to generate issues at low severity levels and display them, e.g., in logs or GitHub PR annotations. They may not always want to fail when issues are reported. Currently, TFLint exits non-zero (2) when any issues are detected.

## Proposal

Allow users to set a minimum severity for exiting with an error code. For example:

```sh
tflint --min-error-severity warning
```

This would cause issues at the `info` severity level to be printed, but TFLint would still return a 0 exit code.

## References

* #1555

**Comment by wata727** [comment]
I don't think it's a bad idea to provide such a feature, but we should be careful with the naming.
For example the `--min-error-severity` has the following ambiguity:

- What do you mean by "error"? Is it different from the issue's "error" severity?
- Are the `info` level issues ignored?

We'll need naming that can resolve these.

**Comment by bendrucker** [comment]
Agreed, `min-failure` would be less ambiguous

**Comment by sergei-ivanov** [comment]
Or `--minimum-severity` like in tfsec.
